discord/template.py

```diff
@@ -197,7 +201,25 @@ async def sync(self):
         data = await self._state.http.sync_template(self.source_guild.id, self.code)
         self._store(data)
 
-    async def edit(self, **kwargs):
+    @overload
+    async def edit(
+        self,
+        *,
+        name: Optional[str] = ...,
+        description: Optional[str] = ...,
+    ) -> None:
+        ...
+
+    @overload
+    async def edit(
+        self,
+        *,
+        name: Optional[str] = ...,
+        description: Optional[str] = ...,
+    ) -> None:
+        ...
```

Comment:
These two are the same.

I know you're doing this because the `overload` decorator requires two overloads for it to work (a bizarre limitation) and this is because kwargs typing is basically impossible.

Unfortunately due to limitations of kwargs typing this means that using kwargs doesn't play well. I wonder if it'd be better to spread the kwargs instead. For this in particular (two parameters) it's not too bad, though for other things it's going to be a disaster.

Alternatively rather than spelling out *every* kwargs twice in both the overload sets, one can just be an empty function: 

```py
@overload
async def edit(self) -> None:
    ...
```

I haven't tested this out but it'd hope it's valid -- even if the construction makes no sense.